hikari/impl/entity_factory.py

```diff
@@ -2752,6 +2834,13 @@ def deserialize_component_interaction(
         if resolved_payload := data_payload.get("resolved"):
             resolved = self._deserialize_resolved_option_data(resolved_payload, guild_id=guild_id)
 
+        authorizing_integration_owners = {
+            application_models.ApplicationIntegrationType(int(integration_type)): snowflakes.Snowflake(
+                int(integration_owner_id)
+            )
```

Comment:
I believe for snowflakes, you do not need to cast it to an int before hand (as Snowflake is a subclass of int)